coalib/bearlib/languages/definitions/g4.py

```diff
@@ -0,0 +1,21 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class g4:
+    aliases = 'antlr4', 'grun'
+    extensions = '.g4',
+    comment_delimiters = '//',
+    multiline_comment_delimiters = {'/*': '*/'}
+    string_delimiters = {"'": "'"}
+    multiline_string_delimiters = {}
+    indent_types = {'{': '}'}
+    encapsulators = {'(': ')', '[': ']'}
+    keywords = [
+        'import', 'fragment', 'lexer',
+        'parser', 'grammar', 'returns',
+        'locals', 'throws', 'catch',
+        'finally', 'mode', 'options',
+        'tokens']
+    special_chars = list(r'/**+-*/.;\,()[]{}\=<>|&^~?%!')
+    string_delimiter_escape = {"'": "\\'"}
```

Comment:
Possibly you meant to escape both the start and end tags for a string delimiter ?
Also this is a new usage, I've not seen it before. Only seen the braces being escaped on a need basis.
Would be great if you could put a link which may expand on this usage as a comment in this PR for future reference :sweat_smile:

Reply:
Yes i meant to escape both the start and end tags for a string delimiter.I followed [this](https://github.com/antlr/antlr4/blob/master/doc/lexicon.md) where both ```char``` and ```string``` use single quote.Also at the very first line they have written that they follow syntax of C so should i include this link at the PR body as taken references from so and so resource.

Reply:
So for escaping both start and end, should'nt it be `{"\\'": "\\'"} ?

Reply:
Alright, got it.

Reply:
But check [this](https://github.com/coala/coala/blob/master/coalib/bearlib/languages/definitions/C.py#L22) for other languages too.

Reply:
Oh
Github escaped my comment
I'd meant to write:
`{"\\'": "\\'"}`

Since a single backslash is equivalent to no backslash
And I'm not quite sure about the one present in c definition, not sure what's the purpose of that, but seems like its an error.